src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -307,9 +307,12 @@ def helm_install_release(chart_path, subscription_id, kubernetes_distro, kuberne
                         "--set", "systemDefaultValues.clusterconnect-agent.enabled=true",
                         "--output", "json"]
     # Add custom-locations related params
-    if enable_custom_locations:
+    if enable_custom_locations and (enable_private_link is None or enable_private_link == "false"):
         cmd_helm_install.extend(["--set", "systemDefaultValues.customLocations.enabled=true"])
         cmd_helm_install.extend(["--set", "systemDefaultValues.customLocations.oid={}".format(custom_locations_oid)])
+    # Disable cluster connect if private link is enabled
+    if enable_private_link == "true":
```

Comment:
Why does this logic judge `enable_private_link` as string type instead of bool type?

Reply:
changed to get_three_state_flag()